### PR TITLE
Doc changes for FASC v2.7

### PR DESCRIPTION
**Pull request by manuwelakanade-GSLAB**
Doc changes for FASC v2.7

### CODE REVIEW

**Comment by manuwelakanade-GSLAB** [comment]
Hi @JosephPJordan : Can you please review and merge this pull request. Thanks.

File: docs/faq.rst [hunk 1]
```diff
@@ -60,6 +60,14 @@ In order to recreate the deleted log files, disable and re-enable the applicatio
 
 The application UI may show the 'Request timeout' error, if the application or APIC is receiving a lot of traffic. You can retry the same operation that displayed the error and it should be successful after one or more retries. 
 
+------
+
+**Q. Why do I see the error "Error from BIG-IP: X-F5-Auth-Token does not exist" on performing a BIG-IP login from FASC app?**
```
Comment (JosephPJordan) [inline docs/faq.rst:65]: ... when performing a BIG-IP login..." is better.

File: docs/faq.rst [hunk 2]
```diff
@@ -60,6 +60,14 @@ In order to recreate the deleted log files, disable and re-enable the applicatio
...
+
+If the version of BIG-IP is changed and a re-login to the BIG-IP is attempted from the FASC app, this error may be seen.
```
Comment (JosephPJordan) [inline docs/faq.rst:67]: ...BIG-IP has changed, and you attempt to re-login to the BIG-IP from the FASC app, you may see this error.

File: docs/faq.rst [hunk 3]
```diff
@@ -60,6 +60,14 @@ In order to recreate the deleted log files, disable and re-enable the applicatio
...
+
+Workaround: Delete the BIG-IP from the FASC app UI and re-login to the BIG-IP again.
```
Comment (JosephPJordan) [inline docs/faq.rst:69]: don't need both 're-' and 'again'.  So "...app UI and login to the BIG-IP again.

File: docs/faq.rst [hunk 4]
```diff
@@ -91,6 +101,40 @@ The F5 ACI ServiceCenter Visibility tables have an option to select the **Partit
...
 
 
+Visibility Dashboard
+--------------------
+
+**Q. Why is the BIG-IP Endpoint Details section on the Visibility Dashboard blank?** (Applicable to v2.7+)
+
+BIG-IP Endpoint Details section on the Visibility Dashboard may not display information due to the MAC address table getting flushed on the BIG-IP.
+
+Workaround: Send an ARP request to the host or check the connectivity with the host using ping command.
```
Comment (JosephPJordan) [inline docs/faq.rst:111]: ... using a **ping** command.  (or 'the ping command')

File: docs/faq.rst [hunk 5]
```diff
@@ -91,6 +101,40 @@ The F5 ACI ServiceCenter Visibility tables have an option to select the **Partit
...
+
+**Q. Why is the 'Interface' column blank in the BIG-IP endpoint details section on the Visibility Dashboard?** (Applicable to v2.7+)
+
+The **Interface** column in the BIG-IP Endpoint Details section on the Visibility Dashboard will be blank for vCMP guests since the behavior of the vCMP Guest BIG-IP also is the same; i.e. no interface information for VLANs.
+
+------
+
+**Q. Why are the pool members displayed on Visibility VIP table, and the pool members displayed on the Visibility VIP dashboard not the same?**
+
+Visibility VIP table displays the pool members from a BIG-IP VIP, only if they are also present on the APIC. However, the Visibility dashboard shows all the pool members (and associated stats) that are present on the BIG-IP even if they may or may not be present on the APIC. Hence both the outputs may be different.
```
Comment (JosephPJordan) [inline docs/faq.rst:123]: The visibility VIP table...

File: docs/faq.rst [hunk 6]
```diff
@@ -91,6 +101,40 @@ The F5 ACI ServiceCenter Visibility tables have an option to select the **Partit
...
+
+------
+
+**Q. What does the field 'Route Domain' on the Visibility Dashboard indicate?**
+
+It displays the **default route domain** for the partition to which the Virtual Server (VIP) or Node belongs to.
```
Comment (JosephPJordan) [inline docs/faq.rst:129]: ... to which the Virtual Server (VIP) or Node belongs.  
<don't need 'to' at the end with "to which" earlier in the sentence>

File: docs/faq.rst [hunk 7]
```diff
@@ -91,6 +101,40 @@ The F5 ACI ServiceCenter Visibility tables have an option to select the **Partit
...
+
+------
+
+**Q. Why don't I see the scrollbar for 'View Logs' window on the Visibility Dashboard?**
```
Comment (JosephPJordan) [inline docs/faq.rst:133]: ...for the 'View Logs' window...

File: docs/faq.rst [hunk 8]
```diff
@@ -91,6 +101,40 @@ The F5 ACI ServiceCenter Visibility tables have an option to select the **Partit
...
+
+If you encounter this issue, please zoom out your web browser window by clicking the **Ctrl** key, and then press the **-** (i.e. minus key).
```
Comment (JosephPJordan) [inline docs/faq.rst:135]: ...this issue, use the 'zoom out' option on your web browser. For example, on Windows, hold the **Ctrl** key, and then click **-** (the dash/minus key).

File: docs/navigate.rst [hunk 9]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
 2. In the top right of the Visibility tab, click :guilabel:`Refresh`.
 
 The contents of the visibility table are refreshed.
+
+
+
+Configure Telemetry
+-------------------
+
+1. To view the Virtual Server statistics on the Visibility VIP Dashboard, Telemetry plugin version 1.17 or higher needs to be installed on the BIG-IP device.
```
Comment (JosephPJordan) [inline docs/navigate.rst:89]: ...VIP Dashboard, you must install the Telemetry Streaming plugin version 1.17 or higher on the BIG-IP device.

File: docs/navigate.rst [hunk 10]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+
+2. Follow the installation steps from https://clouddocs.f5.com/products/extensions/f5-telemetry-streaming/latest/installation.html
+
+3. Configure a default pull consumer. For example: https://clouddocs.f5.com/products/extensions/f5-telemetry-streaming/latest/pull-consumers.html#pull
+
+4. Recommended pull consumer configuration: For a scaled environment it is highly recommended to retrieve just the Pool and Virtual Server stats from the BIG-IP. One way filter the VIP and Pool statistics is to create a default pull consumer configuration via  a POST request to: https://<BIG_IP>/mgmt/shared/telemetry/declare
```
Comment (JosephPJordan) [inline docs/navigate.rst:95]: For a scaled environment, we strongly recommend you retrieve just the Pool...

One way to filter the VIP... <missing the word 'to'>

File: docs/navigate.rst [hunk 11]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+      }
+
+5. If further customization is required to filter only specific virtual servers and pools, refer to this link: https://clouddocs.f5.com/products/extensions/f5-telemetry-streaming/latest/data-modification.html
+ 
+View VIP Dashboard
+-------------------
+
+1. Click the Visibility tab, and then from the Table list, click the VIP table.
+
+2. From the Partition list, click the partition of interest.
```
Comment (JosephPJordan) [inline docs/navigate.rst:133]: ...click the appropriate partition.

File: docs/navigate.rst [hunk 12]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+4. Click the VIP for which the dashboard is to be displayed. This redirects to the Visibility Dashboard sub-tab. 
+   
+   .. note::
+   
+      - It is possible to directly click the **Visibility Dashboard** sub-tab and then select the VIP, instead of a redirect from **Visibility Table**.
+      
+
+5. Visibility Dashboard displays the information for the selected VIP including VIP name, service port, protocol, SNAT, route domain, iRules, default pool (and pool information such as load balancing and pool monitor)
+         
+         a. Dashboard displays a **Telemetry Consumer** drop-down list. Select the appropriate consumer from which the statistics need to be obtained. The Virtual pool statistics will be displayed on the dashboard including Bits, Packets, Connections and Requests.
```
Comment (JosephPJordan) [inline docs/navigate.rst:146]: The dashboard displays...

File: docs/navigate.rst [hunk 13]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+      - It is possible to directly click the **Visibility Dashboard** sub-tab and then select the VIP, instead of a redirect from **Visibility Table**.
+      
+
+5. Visibility Dashboard displays the information for the selected VIP including VIP name, service port, protocol, SNAT, route domain, iRules, default pool (and pool information such as load balancing and pool monitor)
+         
+         a. Dashboard displays a **Telemetry Consumer** drop-down list. Select the appropriate consumer from which the statistics need to be obtained. The Virtual pool statistics will be displayed on the dashboard including Bits, Packets, Connections and Requests. 
+         
+            .. note::
+         
+               - Telemetry plugin needs to be installed on the BIG-IP to be able to view the Telemetry consumer list and stats from the telemetry consumer. Check the **Configure Telemetry** section for installation and configuration.
```
Comment (JosephPJordan) [inline docs/navigate.rst:150]: The Telemetry Streaming...

File: docs/navigate.rst [hunk 14]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+            
+               - If there is only a single Telemetry pull consumer configured, then FASC will directly query that consumer and display the stats.
+            
+         b. **View Stats** - Dashboard displays a **View Stats** link besides the default pool. It displays total pool member count and the availability and enabled state of the pool  members using a pie chart.
```
Comment (JosephPJordan) [inline docs/navigate.rst:154]: The Dashboard... link next to the default pool <or beside, not besides>
It displays the total pool member count...

File: docs/navigate.rst [hunk 15]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+         
+         c. **View Logs** - Dashboard displays a **View Logs** link besides the VIP, which, on click will open a popup window and display all the logs for the selected Virtual Server.
```
Comment (JosephPJordan) [inline docs/navigate.rst:156]: The Dashboard...  link next to the VIP, which, when clicked opens a popup window and displays...

File: docs/navigate.rst [hunk 16]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+         
+         c. **View Logs** - Dashboard displays a **View Logs** link besides the VIP, which, on click will open a popup window and display all the logs for the selected Virtual Server. 
+         
+            .. note::
+
+               - It is possible to also view all logs instead of the filtered logs.
+            
+               - It is possible to specify a UTC from and to Date/Time for the logs to be retrieved. The From/To time should be specified in UTC timezone. The response too will be in UTC.
+
+         d. **View Connections** - Dashboard will display a **View Connections** link which, on click will display all the active connections to that VIP.
```
Comment (JosephPJordan) [inline docs/navigate.rst:164]: The Dashboard displays a **View Connections** link, which, when clicked displays all the active...

File: docs/navigate.rst [hunk 17]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+         
+         e. **APIC Endpoint Details** - Visibility dashboard displays APIC details for the VIP:
```
Comment (JosephPJordan) [inline docs/navigate.rst:166]: The Visibility dashboard...

File: docs/navigate.rst [hunk 18]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+                                        1. MAC 2. EPG 3. Node 4. Interface 5. VLAN Encap
+                                        
+         f. **BIG-IP Endpoint Details** - Visibility dashboard displays BIG-IP details for the VIP:
```
Comment (JosephPJordan) [inline docs/navigate.rst:169]: The Visibility...

File: docs/navigate.rst [hunk 19]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+                                          1. MAC 2. VLAN 3. Interfaces (There is a **View Logs** link besides interfaces to view the interface logs) 4. Self IPs
+
+
+View Node Dashboard
+-------------------
+
+1. Click the Visibility tab, and then from the Table list, click the Node table.
+
+2. From the Partition list, click the partition of interest.
```
Comment (JosephPJordan) [inline docs/navigate.rst:178]: ... click the appropriate partition.

File: docs/navigate.rst [hunk 20]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+
+3. The Node table for this BIG-IP and partition will be displayed.
+
+4. Click the Node for which the dashboard is to be displayed. This redirects to the Visibility Dashboard sub-tab. 
+
+         .. note::
+   
+            - It is possible to directly click the **Visibility Dashboard** sub-tab and then select the Node, instead of a redirect from **Visibility Table**.
+      
+5. Visibility Dashboard displays the information for the selected VIP including Node name, route domain and monitor.
```
Comment (JosephPJordan) [inline docs/navigate.rst:188]: The Visibility Dashboard...

File: docs/navigate.rst [hunk 21]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+
+         a. **View Logs** - Dashboard displays a **View Logs** link besides the Node, which, on click will open a popup and all the logs for the selected Node.
```
Comment (JosephPJordan) [inline docs/navigate.rst:190]: The Dashboard displays.....  link next to the Node, which, when clicked opens a popup...

File: docs/navigate.rst [hunk 22]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+
+            .. note::
+
+               - It is possible to also view all logs instead of the filtered logs.
+            
+               - It is possible to specify a UTC from and to Date/Time for the logs to be retrieved. The From/To time should be specified in UTC timezone. The response too will be in UTC.
+
+         b. **View Connections** - Dashboard displays a **View Connections** link which, on click will display all the active connections to that Node.
```
Comment (JosephPJordan) [inline docs/navigate.rst:198]: The Dashboard displays.... which, when clicked displays all the active...

File: docs/navigate.rst [hunk 23]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+
+         c. **APIC Endpoint Details** - Visibility dashboard displays APIC details for the Node:
```
Comment (JosephPJordan) [inline docs/navigate.rst:200]: The Visibility...

File: docs/navigate.rst [hunk 24]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+                                        1. MAC 2. EPG 3. Node 4. Interface 5. VLAN Encap 
+                                        
+         d. **BIG-IP Endpoint Details** - Visibility dashboard displays BIG-IP details for the Node:
```
Comment (JosephPJordan) [inline docs/navigate.rst:203]: The Visibility

File: docs/navigate.rst [hunk 25]
```diff
@@ -80,3 +80,126 @@ Refresh Visibility tab
...
+                                          1. MAC 2. VLAN 3. Interfaces (There is a **View Logs** link besides interfaces to view the interface logs) 4. Self IPs
```
Comment (JosephPJordan) [inline docs/navigate.rst:204]: "beside" or "next to"

File: docs/release-notes.rst [hunk 26]
```diff
@@ -1,3 +1,233 @@
...
+
+**Workaround**: Wait for 20 minutes and try to access the application again. The Glusterfs should recover automatically and the application should be accessible again.
+
+------
+
+**Export db does not work on APIC version 5.0.1**
+
+The user menu on the top-right corner of the F5 ACI ServiceCenter has an option for exporting F5 ACI ServiceCenter database (**Export/Import DB > Export DB**). This option does not work on APIC versions 5.0.1. This is working as expected on prior versions of the APIC and also on versions 5.0(2e) and higher. 
+
+**Workaround:** Backup the F5 ACI ServiceCenter database via the following steps:
```
Comment (JosephPJordan) [inline docs/release-notes.rst:26]: ...using the following steps:

File: docs/release-notes.rst [hunk 27]
```diff
@@ -1,3 +1,233 @@
...
+
+------
+
+**Visibility table fast scrolling on Mozilla Firefox for scale configs may result in a blank screen**
+
+This is a known ag-grid issue on the Mozilla Firefox browser: https://github.com/ag-grid/ag-grid/issues/2841
+
+**Workaround:** Scroll slowly to prevent this issue. But if this issue is observed, the possible workarounds are:
+
+1. Switch tabs from Visibility tab to one of the other tabs and then switch back.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:77]: ..the Visibility tab

File: docs/release-notes.rst [hunk 28]
```diff
@@ -1,3 +1,233 @@
...
+**Telemetry defect:** https://github.com/F5Networks/f5-telemetry-streaming/issues/108
+
+------
+
+L4-L7 App Services
+------------------
+
+**Application services declaration not deleted**
+
+If your AS3 declaration contains “optimisticLockKey” mentioned explicitly, the Application Services configuration may not be deleted completely, even after multiple attempts from the application UI. However, the configuration gets removed from the BIG-IP device.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:120]: However, the configuration is removed from the BIG-IP device.

File: docs/release-notes.rst [hunk 29]
```diff
@@ -1,3 +1,233 @@
...
+-------
+
+
+Dynamic Endpoint Attach Detach
+------------------------------
+
+
+**Error on EPG mapping delete operation**
+
+When a dynamic endpoint mapping is added to an application using **Manage Endpoint Mappings**, the application gets created on the BIG-IP. If this mapping is deleted using the **RESET** button on **Manage Endpoint Mappings** form, users may encounter an error "The requested Pool Member (/Partition/App/Pool /NodePartition/NodeIP) was not found."
```
Comment (JosephPJordan) [inline docs/release-notes.rst:133]: ..., the application is created on the BIG-IP.

File: docs/release-notes.rst [hunk 30]
```diff
@@ -1,3 +1,233 @@
...
+
+Note: Similar issues might be seen with other erronous configurations such as unsupported IPv4 formats like 1.2.3.4/24 instead of 1.2.3.4
+
+AS3 Defect: https://github.com/F5Networks/f5-appsvcs-extension/issues/287
+
+------
+
+**Pool members deleted or added directly to BIG-IP don't get updated automatically after clicking "Sync EPs".**
+
+1. If BIG-IP pool members automatically get added by the **Dynamic endpoint discovery** feature, but then few endpoints are deleted directly from the BIG-IP (i.e. out of band); these endpoints do not get created again on clicking **L4-L7 App Services --> Application Inventory --> Sync EPs** for that application.
```
Comment (JosephPJordan) [inline docs/release-notes.rst:211]: If BIG-IP pool members are automatically added by the...

**Comment by JosephPJordan** [comment]
Hi @manuwelakanade-GSLAB, a few suggested changes.  Let me know if there's a better way we can do this (ie if i should just make changes directly to the files, etc).

**Comment by manuwelakanade-GSLAB** [comment]
Hi @JosephPJordan : Thanks for the review. I have addressed all your comments and updated the pull request. Can you please merge this request if there aren't any further comments. Once this is merged, I will raise a pull request from staging to master branch. Thank you.

**Review by JosephPJordan — APPROVED** [review]

**Comment by manuwelakanade-GSLAB** [comment]
Thanks @JosephPJordan

**Pull request merged**